src/imitation/data/fragments.py

```diff
@@ -0,0 +1,87 @@
+import random
+import warnings
+from typing import Callable, List, Optional, Tuple
+
+from imitation.data.types import TrajectoryWithRew
+
+Fragmenter = Callable[
+    [List[TrajectoryWithRew]], List[Tuple[TrajectoryWithRew, TrajectoryWithRew]]
+]
+"""Creates pairs of trajectory fragments from a collection of trajectories."""
+
+
+class RandomFragmenter:
+    """Sample fragments of trajectories uniformly at random with replacement.
+
+    Note that each fragment is part of a single episode and has a fixed
+    length. This leads to a bias: transitions at the beginning and at the
+    end of episodes are less likely to occur as part of fragments (this affects
+    the first and last fragment_length transitions).
+
+    TODO(ejnnr): should we correct for this bias and make all transitions equally
```

Comment:
Good question, how would you correct for the bias?

One approach would be to just partition the trajectories into `fragment_length` fragments, and then choose from those fragments randomly. Since no overlap, transitions at the start/end are as likely to occur as those in the middle. But you now have a new bias of never seeing trajectories that cross a certain boundary of timesteps. Also this won't work if episodes aren't a multiple of `fragment_length`.

In general I'm not convinced it's possible to get uniform probability over transitions without adding restrictions to how the trajectories are sliced.